### PR TITLE
Closes #192 | Create dataset loader for MALINDO_parallel

### LINKED ISSUE #192: Create dataset loader for MALINDO_parallel
Dataloader name: `malindo_parallel/malindo_parallel.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?malindo_parallel

|     Dataset| malindo_parallel |
|-------------|---|
| Description | Teks ini adalah skrip video untuk Kampus Terbuka Universiti Bahasa Asing Tokyo pada tahun 2020. Tersedia parallel sentences dalam Bahasa Melayu/Indonesia dan Bahasa Jepang |
| Subsets | - |
| Languages | zlm, ind |
| Tasks | Machine Translation |
| License | Creative Commons Attribution 4.0 (cc-by-4.0) |
| Homepage | https://github.com/matbahasa/MALINDO_Parallel/tree/master/OpenCampusTUFS |
| HF URL | - |
| Paper URL | https://github.com/matbahasa/MALINDO_Parallel/blob/master/README.md |

### PR DESCRIPTION
**Pull request by bryanwilie**
Closes #192.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by MJonibek** [comment]
Please can you add `__init__.py` file, and also remove all comments (especially TODO)

**Comment by bryanwilie** [comment]
Hi @MJonibek, that's just done. Thank you for reviewing!

File: seacrowd/sea_datasets/malindo_parallel/malindo_parallel.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+"""
+
+
+_HOMEPAGE = "https://github.com/matbahasa/MALINDO_Parallel/tree/master/OpenCampusTUFS"
+
+
+_LANGUAGES = ["zlm", "jpn"]  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+
+
+_LICENSE = "Creative Commons Attribution 4.0 (cc-by-4.0)"  # example: Licenses.MIT.value, Licenses.CC_BY_NC_SA_4_0.value, Licenses.UNLICENSE.value, Licenses.UNKNOWN.value
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/malindo_parallel/malindo_parallel.py:63]: Please remove the `# example` comments (also in `_SUPPORTED_TASKS`), 

```suggestion
_LICENSE = "Creative Commons Attribution 4.0 (cc-by-4.0)" 
```

File: seacrowd/sea_datasets/malindo_parallel/malindo_parallel.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+_SUPPORTED_TASKS = [Tasks.MACHINE_TRANSLATION]  # example: [Tasks.TRANSLATION, Tasks.NAMED_ENTITY_RECOGNITION, Tasks.RELATION_EXTRACTION]
+
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+
+class MalindoParallel(datasets.GeneratorBasedBuilder):
```
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/malindo_parallel/malindo_parallel.py:83]:
```suggestion
class MalindoParallelDataset(datasets.GeneratorBasedBuilder):
```

**Review by ljvmiranda921 — COMMENTED** [review]
Just a few small nits but the data loader works properly :)
Please rerun `make check_file` as well to ensure that the formatting is consistent with the other implementations.

**Comment by bryanwilie** [comment]
Sorry for the very late reply. Suggestions are committed, thank you for the feedback!

**Comment by ljvmiranda921** [comment]
LGTM! Let's just wait for @holylovenia 's review :)

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @bryanwilie, thank you for the dataloader! It generally works well, but I find that it returns erroneous data instances for these exact IDs:

```
Train 59
{'id': '59',
 'text_1': 'Selain itu, pada peringkat penghujung tahun satu, hampir semua '
           'pelajar mengikuti Program Kembara Ilmu Inap Desa selama tiga '
           'minggu di Malaysia dan Singapura.',
 'text_1_name': 'zlm',
 'text_2': '',
 'text_2_name': 'jpn'}

Train 60
{'id': '60',
 'text_1': 'また、1年次の終わりにはほとんどの学生がマレーシア・シンガポールでの3週間のホームステイ・プログラムに参加します',
 'text_1_name': 'zlm',
 'text_2': 'また、1年次の終わりにはほとんどの学生がマレーシア・シンガポールでの3週間のホームステイ・プログラムに参加します',
 'text_2_name': 'jpn'}
```

This is due to [the original data instance ID 8-4 is split into 2 lines](https://github.com/matbahasa/MALINDO_Parallel/blob/0881bd2a9d0cd28e65a46c81e1605acdf1c95530/OpenCampusTUFS/OCTUFS2020.txt#L60) rather than the usual 1 line.

Additionally, could you please modify the `id` values to follow the IDs of the original data, e.g., `8-4`, `9-1`, etc.?

**Comment by holylovenia** [comment]
Hi @bryanwilie, please let us know if you need any help for this PR! 👍

**Comment by holylovenia** [comment]
Hi @bryanwilie, please let us know if you need any help for this PR! 👍

**Comment by bryanwilie** [comment]
should be done!

**Review by holylovenia — APPROVED** [review]
LGTM!! Thanks a lot for your help, @bryanwilie!!

**Pull request merged**